loguru.hpp

```diff
@@ -102,6 +102,35 @@ Website: www.ilikebigbits.com
 #include <sal.h>	// Needed for _In_z_ etc annotations
 #endif
 
+/*
+	Will add syslog as a standard sink for log messages
+	Any logging message with a verbosity lower or equal to
+	the given verbosity will be included.
+
+	This works for Unix like systems (i.e. Linux/Mac)
+	There is no current implementation for Windows (as I don't know the
+	equivalent calls or have a way to test them). If you know please
+	add and send a pull request.
+
+	The code should still compile under windows but will only generate
+	a warning message that syslog is unavailable.
+
+	Search for LOGURU_SYSLOG to find and fix.
+*/
+#if defined(__linux__) || defined(__APPLE__)
+#define LOGURU_SYSLOG 1
+#endif
```

Comment:
In Loguru I prefer `#if LOGURU_SYSLOG` to `#ifdef LOGURU_SYSLOG` (because the latter is more error prone). So:

``` C
#if defined(__linux__) || defined(__APPLE__)
   #define LOGURU_SYSLOG 1
#else
   #define LOGURU_SYSLOG 0
#endif
```

Reply:
Done.

But not the indenting.
technically the '#' has to be in column 0 to be standard compliant.

Reply:
Note: I disagree with the error prone comment. This technique is more error prone.

This is why you see include gurads as `#ifdef <GUARD>` and why you check for system with `#ifdef _WIN32` etc.

Reply:
Let me explain why I think `#ifdef` is error prone!

`#ifdef LOGORU_SYSLOG` will compile fine, but won't do what you want it do do.

`#if LOGORU_SYSLOG` on the other hand won't compile, which is good, because you meant to type `#if LOGURU_SYSLOG`. So by changing `#ifdef` to an `#if` the compiler will now catch typos for you! Yay :)